elastic/kibana
# [Search Sessions] Fix restoration URL conflicts with global time sync

## Summary

Fixes https://github.com/elastic/kibana/issues/87149

See reproduction of the bug:

https://user-images.githubusercontent.com/7784120/103542768-b8ab5400-4e9d-11eb-8f7d-1d11baa24c3c.mov

This is "an easy" fix where apps remove `searchSessionId` query param from the URL that is going to be used in chrome's nav links.

Long term we should explore [fetching app state from SO](https://github.com/elastic/kibana/issues/87446) instead of relying on state in the URL for restoring the search session. 

**This pr also fixes a bug in `kbnUrlTracker` where state from initial URL wasn't stored in session storage so it wouldn't appear in nav link.**

### How to test 

This pr is hard to test without https://github.com/elastic/kibana/pull/81707 in real world. 
Bug to validate that the fix works: 

1. Load discover app using a URL with appended fake `&searchSessionId=fake`
2. Discover should loaded with an error 
3. Go to dashboard
4. Check that chrome link to discover app has your URL but without `searchSessionId` part

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
